controllers/machinedeployment_sync.go

```diff
@@ -104,19 +103,22 @@ func (r *MachineDeploymentReconciler) getNewMachineSet(d *clusterv1.MachineDeplo
 	// latest revision.
 	if existingNewMS != nil {
 		msCopy := existingNewMS.DeepCopy()
-		patch := client.MergeFrom(msCopy.DeepCopy())
+		patchHelper, err := patch.NewHelper(msCopy, r.Client)
```

Comment:
Can we get rid of these "inner" patch helpers?

Reply:
I don't think we can safely do that today, unless we have a way to keep track of the old/new MachineSet the MachineDeployment is managing